libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -840,8 +841,9 @@ class MapboxNavigation @VisibleForTesting internal constructor(
      *           versionInfo: RoadGraphVersionInfo?
      *       ) {
      *           if (isUpdateAvailable) {
-     *               mapboxNavigation.onDestroy()
```

Comment:
This had some pretty bad consequences. Considering [onDestroy will remove all registered listener](https://github.com/mapbox/mapbox-navigation-android/blob/bf0f61f96c3942b28daea4aa010238380b20acb3/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L1016-L1024)

When using `MapboxNavigationApp`, this recreation can be done and listeners can be re-registered in `MapboxNavigationObserver.onAttached`